promql/query_logger_test.go

```diff
@@ -110,10 +110,7 @@ func TestMMapFile(t *testing.T) {
 	filename := file.Name()
 	defer os.Remove(filename)
 
-	fileAsBytes, closer, err := getMMapedFile(filename, 2, nil)
-	if err != nil {
-		t.Cleanup(func() { closer.Close() })
-	}
```

Comment:
What I don't understand here is, why not close `closer` at the end of the test? If there's no error, there will be a valid `closer`, right?

Reply:
Good catch, I bet `defer os.Remove(filename)` in 111 is already returning an error sometimes (on windows).

Reply:
I also don't understand FWIW why `file` is not closed immediately, as you are just using the name.

Reply:
his also seems unrelated to the PR. We can address it in that follow-up PR.

Reply:
@machine424 I think this should be changed into 
```gol
	fileAsBytes, closer, err := getMMapedFile(filename, 2, nil)
	if err == nil {
		t.Cleanup(func() { closer.Close() })
	}
```